### ISSUE TITLE
Abilities do not exist in Morph 1.12.2. Can't fly, can't swim, etc. More info in issue.

### ISSUE THREAD

**Issue by iChun** [iChun/Morph#884]
tl;dr No abilities in Morph. It'll be back in the mod at a future date. Issues about this will be closed.

===========================================

If you folks are unaware, Morph 1.12.2 is a full rewrite of the Morph mod that I first wrote 4 years ago. I'm a lot happier with the code base and it's a lot easier to modify or change things under the hood.

However, with a rewrite, it takes a lot more time than porting. Morph's abilities are also slated to be rewritten and I currently don't have the time to reimplement the abilities system and all the old abilities we had before, not until January to the very least.

So if you're looking for all the old abilities, they're not in yet. No flight, no swimming, no climbing etc.

Any issues about morph's abilities will be closed without any notice.

**Comment by PapiDimmi** [comment]
d

**Comment by PapiDimmi** [comment]
Didn’t you say on Reddit you’ll make a better system with meteorites and stuff?

**Comment by iChun** [comment]
I want to get Morph to where it was first before I want to expand or add on to the mod.

**Comment by PapiDimmi** [comment]
You are the hero we need, but not the one we deserve.

**Comment by GamerTygoNL** [comment]
ok im going with metamorph until you add them

**Comment by GeoGeo85** [comment]
It's now APRIL first of 2018!!! So... Will you or will you not at some point re-include abilities in Morph?!
Is it still even foregoing?! Or did you deliberately chose to let it that way, at this point? Like I said, It's now 5 months ad a half LATER and, well... You know?

**Comment by iChun** [comment]
@GeoGeo85 Indeed, it is now April first of the year 2018.

I will at some point re-include abilities in Morph, yes. 

Is it still forgoing, yes, bit by bit: https://github.com/iChun/Morph/commit/8da3df704b45cf2f2b6e2c0f81d84e61f5d0b589

Did I deliberately choose to let it this way 5 months and a half later?  If you are unaware, I am currently a fourth year medical student and in the past 5 months, I have had to go through a forensics and legal medicine rotation which exposures us students to actual cases of death, murder, physical and sexual abuse (of both adult and child), etc, and I am currently in psychiatry, which involves seeing and talking to people who are depressed, manic, anxious, delirious, hearing voices, attempted suicide, etc, and digging deep into their past to try and figure out why they have become like this to learn how we can help them to the best of our ability.

We don't get breaks. Our breaks are the weekends where we are expected to study the material for the next week. At this point I'm required to know a ~460 page book practically word for word with a proper understanding which I was only handed 6 weeks ago. With that said, I hope you can understand that when I do have free time, I would like to relax and socialise more rather than sit in front of a screen and strain my brain even more.

I've been making mods for Minecraft for 7 years now, and times change. I still want to make mods and improve on the ones I have out, but it is hard to find motivation to do so when my motivation and primary focus right now is trying to get through med school without turning into one of their patients. The students here are burnt out and the institution knows about it but can't do anything about it. And guess what? It doesn't get easier. In a couple months time, I will be a final year student and I will be expected to be as competent as an actual intern in the hospital.

Anyway, back to your question. Did I deliberately choose to leave the mod in an unfinished/uncompleted state for more than 5 months? Yes, I did. Did I also deliberately choose to leave the mod open source so people such as yourself can learn and possibly contribute your own changes to add more to the mod whilst I am unable? Yes, I did.

I hope I've been able to answer your questions sufficiently. If you have any additional questions, do let me know. Else, I'll leave you with this picture of a pestle (from a pestle and mortar set) being somewhere it doesn't belong. 
![screenshot_20180321-013102_01](https://user-images.githubusercontent.com/2298444/38172517-11dcc93e-35e0-11e8-900b-5e6be6e03e81.jpg)


Cheers

**Comment by Ozzymud** [comment]
Dude, no worries, I'm sure there are many more out there who feel the way I do. Your going into a pretty noble profession and while I can't even begin to fathom the amount of work it takes, I'm pretty sure it is close to super human and deserves your full attention. MC will likely be here when you get back into it or as you said, someone with more knowledge then myself will do what needs to be done here in the meantime.

**Comment by Rudnt2** [comment]
Hello iChun, 
First thanks for this incredible mod. I use it for 2 years and i love it. But the abilities are note disponible yet. So can we know the date, that they will be avaible ? Best regards
Rudnt2

**Comment by Inferbier91** [comment]
I see that you are busy.
but i ust want to say that you are an amazing mod maker so i think your mods is the best Morph mod ever.
where other morph mods you do not have to work for anything to morph(Which is too boring).
Your mod is challenging in getting morph's.
i thank you for making such an amazing mod!❤️

**Comment by CoolDude7014** [comment]
iChun when ever you are done with **ALL** the abilities, send me a DL link

**Comment by frosty4frisk** [comment]
I got this mod and inventory pets and it works just like you had abilities except you need to put them in your inv and hold them
basically you can morph into a spider take a spider inventory pet and climb walls 🤔

**Comment by CoolDude7014** [comment]
oh ty :)

**Comment by whiteboxgaming** [comment]
What is the point of this mod then if you cant use their abilities? I'm using 1.12 and I can morph but no abilities. How did you improve it? Backend code that does not apply and thats pointless if the mod is worthless now? oh that improvement I see good job on screwing it up

**Comment by MCYayes** [comment]
I don't think whiteboxgaming has a point. They're just too young and/or inexperienced to comprehend your situation. Well, either that or they're a sociopath that lacks the basic ability to empathize with others.

iChun, I've admired your work for years and still do. Considering your current work load only makes me appreciate your efforts that much more. I'm gratefully surprised you were able to get it working in any capacity for 1.12.2.

I look forward to any future changes, and if I find the time and ability to look thru the source code I might see if there's anything I can do to help. I haven't made any MC mods, but I've done a fair amount of scripting and code work for other things that I might be able to muddle thru some.

I can't promise anything, but if I can and am able to make something work, I'll message you here about it before doing anything other than using it myself.

Thanks again, and I hope your final year has started well!

**Comment by Redd56** [comment]
How the fuck did that pastel get there

**Comment by MCYayes** [comment]
> I think it's fair to be a little annoyed that a year later and still no Morph abilities. Why is this mod still being included in popular packs??

I can't answer for everyone, but I include it in the mod packs I play along with other configurable mods like Everlasting Abilities so I can have abilities that way but appearances using Morph. 

Appearances are the main reason for using the mod currently. If all you want are the abilities, then use a mod like Everlasting Abilities for that. but if you like being able to collect morphs and shift into those creatures, then use Morph.

If you want both, then for now use both mods. If that's too much effort, then perhaps we can consider "too much effort" to be an understandable reason the abilities within Morph are taking a while.

**Comment by StinkyTACO** [comment]
hi sorry for posting but do the abilities work yet?

**Comment by VGHS-lucaruby** [comment]
the latest version is still the one from 2017 with no morphs :( but if you check the repository the last time something was changed was 5 days ago so... hope?

**Comment by epicgameruk** [comment]
iChun its 2019 when will you add the abilities back to the morph mod? the last update was 2017 please add them back.

**Comment by ThaExile** [comment]
Dude must have just given up on it...its sad really, dude creates a fantastic mod, proceeds to destroy what everyone loved so much about it...and disappears...

**Comment by Krominicon** [comment]
> 
> 
> Dude must have just given up on it...its sad really, dude creates a fantastic mod, proceeds to destroy what everyone loved so much about it...and disappears...

he did leave it open source, and he's already explained why he isn't able to continue developing it. If you really want it that badly then maybe you should learn how to program the mod yourself. Which is something me and my friend are potentially thinking about. I personally think this is a great learning opportunity for other people in the community.

**Comment by Redd56** [comment]
But i procrastinate too much
X3

**Comment by Reploidsham** [comment]
@Krominicon : That'd be great if the mod weren't so big. Trying to "learn" from this would likely be akin to vandalism if we tried to push to this, at least, and we'd still need people to know about it if it DOES work, which would likely require @iChun to participate in the conversation again.

I want abilities too, and @MCYayes offered probably the best interim solution. Metamorph is also a valid one, but that runs into issues if you mimic a race with, say, poison area of effect damage. Your friends won't really like you much.

**Comment by Nowno** [comment]
Ayye bro its 2019, i know how hard it is but please bruv can you add it back

**Comment by Redd56** [comment]
:man_facepalming:

**Comment by Novocastrian** [comment]
> 
> 
> Dude must have just given up on it...its sad really, dude creates a fantastic mod, proceeds to destroy what everyone loved so much about it...and disappears...

I think that many people forget (or simply do not know) that they don't sign an SLA ([Service Level Agreement](https://en.wikipedia.org/wiki/Service-level_agreement)) detailing the turnaround on bug fixes. Creating and maintaining Minecraft mods is everyone's side-hustle at best and their hobby at worst thus is not something they work full- or part-time on. If you want to help, I'm sure iChun would appreciate the assistance!

**Comment by MatthewBruhLol** [comment]
Hey iChun! Just commenting to ask how it's going.

**Comment by Redd56** [comment]
@novacastrian not always. I think sonic ether only does his shaders and is completely relient on his patreon. Some people can do it

**Comment by TehHyperGamer** [comment]
Does that mean the config "HostileAbilityMode" doesn't work?

**Comment by Skii7niX** [comment]
@iChun, I absolutely adore the mod and the work you have done so far and appreciate all the effort you have put into it to make it the best most it can be. Don't let other people get into your head about this. Med school and your career is top priority and working on the mod is next to none. I'm in the military so I understand the struggle. Good luck in the future man!

**Comment by BlueStriker101** [comment]
It is now 2020. Three years since the last update. I'm gonna assume this mod has been abandoned and switch to a better functioning one that won't clutter my modpack with uselessness. I understand the "situation" but it has been THREE years.

**Comment by iChun** [comment]
@BlueStriker101 🤷‍♂️ I've been busy.

**Comment by Redd56** [comment]
Fair enough keep doin whatever your doing we appreciated all you had done already

**Comment by mrmalef** [comment]
Hey iChun! Would this be different if you were being paid? I'd honestly contribute.. The morphing mod adds so much ability to use (Squid in water, bat while traveling, etc.), but not having any abilities makes the mod pretty much useless. The only use I have now is turning into a bat to fit into 1 by 1 holes, but not much else- I know you're busy and things came up, but I'm willing to contribute and I'm sure a lot of other people are too to get this mod updated.

**Comment by iChun** [comment]
@mrmalef thanks, but unfortunately the issue isn't of a financial origin. For the most part it's been time and motivation. I'm currently in final year of Medical School and you could only just imagine how much load and pressure we have to go through for classes. I've a bit of a breather with things being on lockdown (because we're still students we aren't being frontlined), so I've been able to procrastinate studying by spending some time on my mods but that can only persist for so long.

Working on mods and porting them takes a lot more time than people think. It's been relatively slow going as a lot has to be rewritten since I've jumped from 1.12 to 1.15, so for the most part the code I'm writing is close to brand new. And coding takes time, which circles back to the original issue: I don't have that much of it.

Once I've graduated I'll have a bit more time (as well as no more school to think about), and I am more open to working on my mods or playing games. We'll see the situation, then.

**Comment by mrmalef** [comment]
Hey man take all the time you need and thanks for the quick reply! Especially in times like these you need to do what it takes to stay sane and safe and overall going into the medical field is a much bigger priority than Minecraft ever will be. Thank you again for a great mod, I wasn't trying to bash you or anything- We can wait!!! Don't let anyone else try and make you feel bad or anything.

**Comment by pilotsarchive** [comment]
I don't understand all the disrespectful messages, stop crying about everything and wait like everybody else.
If I got such comments, I'd just abandon the mod, you guys are unbelievably childish.
Just let the author do what they want and stop being such a little bitch about everything.
Stop pretending to have any valid reason or any right to treat an author or anybody like this and fuck off.

@iChun I am sorry that you have to read some of this crap on here, holy shit.

**Comment by Reploidsham** [comment]
Sucking butts is also not really helpful, dude. There is a limit to
the amount of expectations someone can have and there hasn't really
even been an update to the situation that'd assuage concerns at all.

When you start toadying this hard, development can and often does
usually stop dead. The only reason it hasn't totally been taken up is
because most of us are novice at best coders and there is already
promising branches such as Changeling that people need to consider
looking at.

**Comment by iChun** [comment]
@Reploidsham I am busy IRL. I do this in my free time. I am not making any promises of updating, I've only expressed that I do want to. I shouldn't be under any obligation to give you all updates on something I do for fun and am not paid to do so. I'm sorry if what I've said hasn't exactly made you sated you much. Welcome to being disappointed.

I've stated very very clearly that my life comes first. My degree comes before my hobby. I appreciate what mrmalef has said because it shows he understands where I'm coming from. The code is all here. iChunUtil, in another branch. My main line is not, and has never been, coding. Stating most of you are "novice at best" is no excuse. If you want to see mods like these have continued support, why not you stop whining and get up and do something about it.

**Comment by Reploidsham** [comment]
... I did mention there were active alternatives, and specifically ego
fellatio is not a good ideal. Again, when I said updates, I meant
updates to whatever you were doing, not necessarily the mod.

**Comment by MCYayes** [comment]
Reploidsham, it seems you aren't comprehending, so I'll chime in and see if
I can get this concept thru to you:

iChun doesn't OWE, you, me, the community, etc ANYTHING AT ALL.
To think otherwise is to flaunt your entitlement.

You (like many of us) have enjoyed iChun's hobby which was generously
shared with the community, or you wouldn't care enough to be commenting at
all.
Now if you still aren't getting the message here, then perhaps you should
excuse yourself from the conversation.
That is all.

**Comment by AngeloDeveloper** [comment]
hey man, i just came here to say that you marked my childhood and at that time i would never think i could talk to you one day, recently i went back to play minecraft and then i find your mod again, even incomplete i can still manage my friends a lot fun, thanks for the experience and good luck in college, a big hug from your Brazilian players ..

**Comment by iChun** [comment]
@Reploidsham I think my ego's just fine regardless of whatever anyone says. It seems pretty balanced out by the people commenting.

Now, for updates on things I do, there's a place and time for everything and GitHub is not the place for that. If you pulled your head out of whatever crevice you shoved it in you might have been able to find my Twitter or my Twitch, both which have been around for years, and which I am regularly active on.

Cheers!

@MCYayes @AngeloDeveloper Thank you so much for your support! :)

**Comment by Ichun-is-a-bitch** [comment]
Ichung you stupid NIGGER FUCKING UPDATE your shitty fucking Mod. Dumb Bitch.

**Comment by kdog0598** [comment]
@Ichun-is-a-bitch Hey man I want the mod updated as much as the next guy but you think this is the way to go about it? You think the one piece of motivation iChun was missing was derogatory slurs and insults? If you are that desperate for a morph mod use metamorph or changeling instead of shouting on github.

**Comment by Ichun-is-a-bitch** [comment]
> @Ichun-is-a-bitch Hey man I want the mod updated as much as the next guy but you think this is the way to go about it? You think the one piece of motivation iChun was missing was derogatory slurs and insults? If you are that desperate for a morph mod use metamorph or changeling instead of shouting on github.

Shut up cunt, you're mad you didn't think of making a throwaway account to insult him first. Jealous sand-nigger alert.

**Comment by Ichun-is-a-bitch** [comment]
> @Ichun-is-a-bitch Hey man I want the mod updated as much as the next guy but you think this is the way to go about it? You think the one piece of motivation iChun was missing was derogatory slurs and insults? If you are that desperate for a morph mod use metamorph or changeling instead of shouting on github.

Stupid fucking Towelhead.

**Comment by Redd56** [comment]
How to downvote haters on GitHub
Google how do I downvote users on GitHub
Google!
Stupid thing!

**Comment by Redd56** [comment]
Also @iChun nice to see you active when ya can be hope your life is going well

**Comment by alexgurrola** [comment]
@iChun Wow, you have a lot on your plate.  When I get some extra time this week, I'll be sure to fork this and help out however I can.

**Comment by alexgurrola** [comment]
@iChun Actually, someone else beat me to it: https://github.com/iChun/Morph/pull/1070

**Comment by iChun** [comment]
Following the PR #1070 , I've already reviewed it and tested it out briefly and seems solid. Abilities are now available on [CurseForge in v7.2.0](https://www.curseforge.com/minecraft/mc-mods/morph/files/all). 

Huge kudos to @Ajadaz for this update. Closing as this has now been resolved.

**Comment by frostblooded** [comment]
@iChun I am sorry about all the people here that think you owe them anything.

Me and my brother love this _**free**_ mod. We are blessed to have you make this mod for us and thank you for the update today.

Keep doing all your medical stuff and don't worry about some mod for some game. We are blessed to have what this mod has right now. <3

**Comment by Pardenia** [comment]
Hello iChun, I came here just to thank you for your work, your mod is my favorite (along with Mo Creatures). I read many comments demanding more and more, they should be grateful for you even doing what you do (both the mod and the med school). So thank you for all, and keep this awesome work going <3

**Comment by BistDuEsTels** [comment]
Are the Abilities on the other mod versions functional? Also is there a way to tip you or something? I haven't read everything in this thread, but there was something about financial problems, so do you have a Patreon or anything of this sort?

**Comment by Natedogwoooof** [comment]
Is there any way that we can tip you

**Comment by Natedogwoooof** [comment]
I have oribkomz with mental health too anger issues bc I'm autistic
